### PR TITLE
feat/154: add a Podcast Platforms Block

### LINKED ISSUE #154: Block: Podcast Services
### Is your enhancement related to a problem? Please describe.

Many podcast sites like to promote all the various podcast services where they syndicate their content (e.g., Pocket Casts, Apple Podcasts) on their homepage and within individual episode posts.  There's currently no way to do this in the core Social Icons block nor any other quick way to do this using Simple Podcasting.  So I propose a block that would allow a site owner to enable/disable the various podcast services and which would provide a visual icon for each of those enabled services.  Open to what we call the block, "Podcast Services" doesn't feel quite right but is the best I can quickly come up with right now.

The following appear to be a good listing of podcast services to include:
- Pocket Casts
- Apple Podcasts
- Google Podcasts
- Stitcher
- PlayerFM
- Overcast
- Pandora
- Castro
- TuneIn
- Spotify
- RSS (for the actual RSS feed of the podcast to be subscribed to in some other manner)

We'll probably want to include these services as new fields in the [podcast settings](https://github.com/10up/simple-podcasting/blob/develop/.wordpress-org/screenshot-3.png) so that the links to the podcast on each service is stored there and called into the block display for actual icon links.

Here's a good listing of the icons to use for each service (see "Player Icon Set"): https://www.podpage.com/badges/.  As shown on the [Apple Podcasts Identity Guidelines page](https://www.apple.com/itunes/marketing-on-podcasts/identity-guidelines.html#apple-podcasts-icon), it might be nice to have black icons and white icons created from the full color versions on podpage.com and allow a site owner to chose from a full color, black, or white icon set for the display in the block.

### Designs

Example icon display from [WP Briefing](https://wordpress.org/news/2022/02/episode-25-five-cents-on-five-for-the-future/):
Screen Shot 2022-03-02 at 1 37 29 PM

### Describe alternatives you've considered

Await initial inclusion of these services in the core Social Icons block: https://github.com/WordPress/gutenberg/issues/36755.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #154 

### Description of the Change

This PR adds the Podcast Platforms block.

#### Podcasting Platforms block

Screenshot 2023-08-06 at 10 26 36 PM

#### Podcasting Platforms settings

Screenshot 2023-08-06 at 10 24 43 PM

### Verification Process

1. Add few podcasts.
2. On the podcasts edit screen, scroll down to the **Podcasting Platforms** section.
3. Set few platforms on which the podcast will be available.
4. Add the Podcast Platforms block to a page and select the podcast from the search field.
5. Confirm the output on the frontend, and also validate a11y.

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Credits

Props @jeffpaul @Sidsector9

### CODE REVIEW

**Comment by Sidsector9** [comment]
@iamdharmesh this is ready for your review.

File: simple-podcasting.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ function podcasting_edit_term_enqueues( $hook_suffix ) {
 		);
 	}
 
-	if ( 'admin_page_simple-podcasting-onboarding' === $hook_suffix ) {
+	if ( in_array( $hook_suffix, $screens, true ) ) {
```
Comment (iamdharmesh) [inline simple-podcasting.php:124]: `admin_page_simple-podcasting-onboarding` should be added to the `$screens` array.
```suggestion
        $screens[] = 'admin_page_simple-podcasting-onboarding';
	if ( in_array( $hook_suffix, $screens, true ) ) {
```

File: assets/js/blocks/podcast-platforms/edit.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	}, [ showId ] );
+
+	/**
+	 * Handler to set the attribute showId.
+	 *
+	 * @param {Int} termId The show ID.
+	 * @returns void
+	 */
+	const onShowSelect = ( termId ) => {
+		setAttributes( { showId: termId } );
```
Comment (iamdharmesh) [inline assets/js/blocks/podcast-platforms/edit.js:130]: Maybe close the search list on select a Show and show a message if no podcast platforms are there for the selected show? For example: In the attached video, "TEST PODCAST" doesn't have any podcast platforms and selecting it not updates the UI and nothing happens on it. Maybe we should show a message like "No podcast platform message found". what do you think?

https://github.com/10up/simple-podcasting/assets/10613171/dafbac61-b1d6-4883-ad70-5420c8aab354

File: includes/datatypes.php [hunk 3]
```diff
@@ -661,6 +820,30 @@ function get_meta_fields() {
 			'type'    => 'select',
 			'options' => get_podcasting_categories_options(),
 		),
+		array(
+			'slug'  => 'podcasting_platforms',
+			'title' => __( 'Podcasting Platforms', 'simple-podcasting' ),
+			'type'  => 'platform_fields',
+		),
+		array(
+			'slug'    => 'podcasting_icon_theme',
```
Comment (iamdharmesh) [inline includes/datatypes.php:829]: Nitpick:
I think keeping the icon theme option at block level is more user-friendly. Not a major issue, but love to hear your thoughts on this.

**Review by iamdharmesh — CHANGES_REQUESTED** [review]
Thank you for the PR and great work here @Sidsector9. Just added 2-3 minor notes to discuss other wise all good.

Thanks

File: includes/datatypes.php (hunk 3 above)
Comment (Sidsector9) [reply includes/datatypes.php:829]: @iamdharmesh that's a good idea, but something we can iteratively implement?

File: assets/css/podcasting-onboarding.scss [hunk 4]
```diff
@@ -2,10 +2,6 @@
 	box-sizing: border-box;
 }
 
-#wpcontent {
-	padding-left: 0;
-}
```
Comment (iamdharmesh) [inline assets/css/podcasting-onboarding.scss]: Minor: Need to fix this space issue in the onboarding header.

Screenshot 2023-08-18 at 12 44 47 PM

File: includes/datatypes.php (hunk 3 above)
Comment (iamdharmesh) [reply includes/datatypes.php:829]: Sure @Sidsector9, maybe we should create a follow-up issue for this?

**Review by iamdharmesh — APPROVED** [review]
Thank you for the changes @Sidsector9, LGTM!

**Pull request merged**